src/test/java/test/configuration/BeforeMethodWithGroupFiltersTest.java

```diff
@@ -0,0 +1,18 @@
+package test.configuration;
+
+import org.testng.annotations.Test;
+import test.InvokedMethodNameListener;
+import test.SimpleBaseTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BeforeMethodWithGroupFiltersTest extends SimpleBaseTest {
+
+    @Test
+    public void beforeMethodWithBeforeGroupsShouldOnlyRunBeforeGroupMethods() {
+        InvokedMethodNameListener nameListener = run(new Class[] { BeforeMethodWithGroupFiltersSampleTest.class });
+        assertThat(nameListener.getInvokedMethodNames())
+                .isEqualTo(BeforeMethodWithGroupFiltersSampleTest.EXPECTED_INVOCATIONS);
```

Comment:
To check, but I think `isEqualTo` is not strict enough and/or won't provide the best feedback in case of wrong assertion. `containsExactly` looks better.